src/components/structures/RoomView.js

```diff
@@ -409,6 +419,15 @@ module.exports = React.createClass({
         }
     },
 
+    onRoom: function(room) {
```

Comment:
If this exists only to handle the case where we join the room after the RoomView is mounted, I'd like that to be spelt out in comments.

I'd also like the comments in `onJoinButtonClicked` to make it clear that it is `onRoom` that is responsible for intialising the timeline - and I'd like to understand whether we ought also to be calling `_initTimeline` there, or if the `setState` there is redundant.